src/ValidatorKeys.cpp

```diff
@@ -60,8 +60,8 @@ ValidatorKeys::ValidatorKeys (
     , secretKey_ (secretKey)
     , tokenSequence_ (tokenSequence)
     , revoked_ (revoked)
+    , publicKey_ (derivePublicKey(keyType_, secretKey_))
```

Comment:
And now because of the `keyPair_` added in the first comment, you're going to need to initialize these differently, too.
```
, keyPair_ (secretKey, derivePublicKey(keyType_, secretKey))
, secretKey_ (keyPair_.first)
, publicKey_ (keyPair_.second)
, tokenSequence_ (tokenSequence)
, revoked_ (revoked)
```
(Also initialize sk and pk together.)